### PR TITLE
xkcdpass: py2 -> py3, add manpage and run tests

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Drive-by cleanup

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/xkcdpass/default.nix [hunk 1]
```diff
@@ -1,25 +1,32 @@
-{ lib, stdenv
-, buildPythonPackage
-, fetchPypi
+{ lib
+, python3Packages
+, installShellFiles
 }:
 
-buildPythonPackage rec {
+python3Packages.buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xkcdpass/default.nix:6]:
```suggestion
buildPythonPackage rec {
```

File: pkgs/development/python-modules/xkcdpass/default.nix [hunk 2]
```diff
@@ -1,25 +1,32 @@
-{ lib, stdenv
-, buildPythonPackage
-, fetchPypi
+{ lib
+, python3Packages
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xkcdpass/default.nix:2]:
```suggestion
, pythonPackages
, fetchPypi
, isPy27
, pytestCheckHook
```

File: pkgs/development/python-modules/xkcdpass/default.nix [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
+, installShellFiles
 }:
 
-buildPythonPackage rec {
+python3Packages.buildPythonPackage rec {
   pname = "xkcdpass";
   version = "1.17.6";
 
-  src = fetchPypi {
+  src = python3Packages.fetchPypi {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xkcdpass/default.nix:10]:
```suggestion
  src = fetchPypi {
```

File: pkgs/development/python-modules/xkcdpass/default.nix [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
-, fetchPypi
+{ lib
+, python3Packages
+, installShellFiles
 }:
 
-buildPythonPackage rec {
+python3Packages.buildPythonPackage rec {
   pname = "xkcdpass";
   version = "1.17.6";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xkcdpass/default.nix:8]:
```suggestion
  version = "1.17.6";
  disabled = isPy27;
```

File: pkgs/development/python-modules/xkcdpass/default.nix [hunk 5]
```diff
@@ -1,25 +1,32 @@
...
     inherit pname version;
     sha256 = "0ghsjs5bxl996pap910q9s21nywb26mfpjd92rbfywbnj8f2k2cy";
   };
 
-  # No tests included
-  # https://github.com/redacted/XKCD-password-generator/issues/32
-  doCheck = false;
+  nativeBuildInputs = [ installShellFiles ];
+
+  checkInputs = with python3Packages; [ pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xkcdpass/default.nix:17]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Using python3Packages in python-modules is not allowed because it breaks multiple python versions. Use ``disabled = isPyXX;`` instead.

**Review by dotlambda — CHANGES_REQUESTED** [review]
You need to change `pkgs/top-level/all-packages.nix`:
```diff
- xkcdpass = with pythonPackages; toPythonApplication xkcdpass;
+ xkcdpass = with python3Packages; toPythonApplication xkcdpass;
```

**Comment by peterhoeg** [comment]
Thanks for the input guys.

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build xkcdpass

File: pkgs/development/python-modules/xkcdpass/default.nix [hunk 6]
```diff
@@ -12,14 +14,21 @@ buildPythonPackage rec {
...
+
+  postInstall = ''
+    installManPage *.?
+    install -Dm444 -t $out/share/doc/${pname} README*
+  '';
 
   meta = with lib; {
-    homepage = "https://pypi.python.org/pypi/xkcdpass/";
     description = "Generate secure multiword passwords/passphrases, inspired by XKCD";
+    homepage = "https://pypi.python.org/pypi/xkcdpass/";
```
Comment (dotlambda) [inline pkgs/development/python-modules/xkcdpass/default.nix:30]: Moving `homepage` seems unintentional.

**Pull request merged**